integration/operator_context_test.go

```diff
@@ -2,46 +2,120 @@ package integration
 	"sigs.k8s.io/e2e-framework/pkg/envconf"
 	"sigs.k8s.io/e2e-framework/pkg/features"
 )
 
-const retryOperationTimeout = 10 * time.Minute
+const (
+	testApiDef      = "test-http"
+	testOperatorCtx = "mycontext"
+)
 
 func TestOperatorContextCreate(t *testing.T) {
 	opCreate := features.New("Operator Context").
-		Assess("Create", func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+		Setup(func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
 			testNS := ctx.Value(ctxNSKey).(string)
 			is := is.New(t)
 
+			// create operator context
+			opCtx, err := createTestOperatorContext(ctx, testNS, envConf)
+			is.NoErr(err) // failed to create operatorcontext
+
 			// create api definition
-			_, err := createTestAPIDef(ctx, testNS, envConf)
+			_, err = createTestAPIDef(ctx, testNS, func(apiDef *v1alpha1.ApiDefinition) {
+				apiDef.Spec.Context = &model.Target{
+					Name:      opCtx.Name,
+					Namespace: opCtx.Namespace,
+				}
+			}, envConf)
 			is.NoErr(err) // failed to create apiDefinition
 
-			_, err = createTestOperatorContext(ctx, testNS, envConf)
-			is.NoErr(err) // failed to create operatorcontext
+			// create api defintion with empty namespace for contextRef
+			_, err = createTestAPIDef(ctx, testNS, func(apiDef *v1alpha1.ApiDefinition) {
+				apiDef.Name = "empty-ns"
+				apiDef.Spec.Context = &model.Target{Name: opCtx.Name}
+				apiDef.Spec.Proxy.ListenPath = "/empty-ns"
+			}, envConf)
+
+			is.NoErr(err) // failed to create apiDefinition
+
+			return ctx
+		}).
+		Assess("context status is updated",
+			func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+				testNS := ctx.Value(ctxNSKey).(string) //nolint:errcheck
+				client := envConf.Client()
+				is := is.New(t)
+
+				opCtx := v1alpha1.OperatorContext{ObjectMeta: metav1.ObjectMeta{Name: testOperatorCtx, Namespace: testNS}}
+
+				err := wait.For(conditions.New(client.Resources()).ResourceMatch(&opCtx, func(object k8s.Object) bool {
+					operatCtx := object.(*v1alpha1.OperatorContext) //nolint:errcheck
+
+					// only one apidef will get linked
+					// other one has empty namespace
+					if len(operatCtx.Status.LinkedApiDefinitions) != 1 {
+						return false
+					}
+
+					if operatCtx.Status.LinkedApiDefinitions[0].Namespace != testNS ||
+						operatCtx.Status.LinkedApiDefinitions[0].Name != testApiDef {
+						return false
+					}
+
+					return true
+				}))
+				is.NoErr(err)
+
+				return ctx
+			}).Assess("apidef was created in dashboard",
+		func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+			is := is.New(t)
 
-			err = retryOperation(retryOperationTimeout, reconcileDelay, func() error {
+			err := wait.For(func() (done bool, err error) {
 				resp, getErr := http.Get("http://localhost:7000/httpbin/get")
 				if getErr != nil {
 					t.Log(getErr)
-					return getErr
+					return false, nil
```

Comment:
Here, `getErr` is not returned in order to prevent breaking the reconciliation loop, right?

Reply:
@buraksekili Function `wait.For` expects function with signature ` func() (done bool, err error)` . Have to return bool there.